spec/lib/castle/api/request_spec.rb

```diff
@@ -4,57 +4,69 @@
   describe '#call' do
     subject(:call) { described_class.call(command, api_secret, headers) }
 
-    let(:http) { instance_double('Net::HTTP') }
-    let(:request_build) { instance_double('Castle::API::Request::Build') }
+    let(:session_sharer) { instance_double('Castle::API::SessionSharer') }
+    let(:session) { instance_double('Net::HTTP') }
     let(:command) { Castle::Commands::Track.new({}).build(event: '$login.succeeded') }
     let(:headers) { {} }
     let(:api_secret) { 'secret' }
+    let(:request_build) { {} }
     let(:expected_headers) { { 'Content-Type' => 'application/json' } }
 
     before do
-      allow(described_class).to receive(:http).and_return(http)
-      allow(http).to receive(:request).with(request_build)
-      allow(Castle::API::Request::Build).to receive(:call)
-        .with(command, expected_headers, api_secret)
-        .and_return(request_build)
+      allow(Castle::API::SessionSharer).to receive(:instance).and_return(session_sharer)
+      allow(session_sharer).to receive(:session).and_return(session)
```

Comment:
there's a lot of mocking here. Can we build our own abstraction for handling HTTP connections and mock it instead? Then, we could only assert on that mocked object instead of mocking `SessionSharer`, `session_sharer` and `session`.

Reply:
good idea, but I would like to limit changes to minimum here